com.unity.cinemachine/Runtime/Behaviours/CinemachineFreeLook.cs

```diff
@@ -434,44 +434,83 @@ float GetYAxisClosestValue(Vector3 cameraPos, Vector3 up)
                 Vector3 flatDir = dir; flatDir.y = 0;
                 if (!flatDir.AlmostZero())
                 {
-                    float angle = Vector3.SignedAngle(flatDir, Vector3.back, Vector3.up);
+                    float angle = UnityVectorExtensions.SignedAngle(flatDir, Vector3.back, Vector3.up);
                     dir = Quaternion.AngleAxis(angle, Vector3.up) * dir;
                 }
                 dir.x = 0;
 
-                // Sample the spline in a few places, find the 2 closest, and lerp
-                int i0 = 0, i1 = 0;
-                float a0 = 0, a1 = 0;
-                const int NumSamples = 13;
-                float step = 1f / (NumSamples-1);
-                for (int i = 0; i < NumSamples; ++i)
+                // We need to find the minimum of the angle of function using steepest descent
+                return SteepestDescent(0f, 1f, dir, cameraPos);
+            }
+            return m_YAxis.Value; // stay conservative
+        }
+
+        float SteepestDescent(float min, float max, Vector3 desiredDirection, Vector3 cameraPos)
+        {
+            const int maxIteration = 10;
+            const float epsilon = 0.00005f;
+            var x = InitialGuess();
+            for (var i = 0; i < maxIteration; ++i)
+            {
+                var angle = AngleFunction(x);
+                var slope = SlopeOfAngleFunction(x);
+                if (slope == 0 || Mathf.Abs(angle) < epsilon)
+                    break; // found best
+                x = Mathf.Clamp(x - (angle / slope), min, max); // clamping is needed so we don't overshoot
+            }
+            return x;
+
+            // localFunctions
+            float AngleFunction(float input)
+            {
+                var point = GetLocalPositionForCameraFromInput(input);
+                return UnityVectorExtensions.SignedAngle(desiredDirection, point, Vector3.right);
+            }
+            // approximating derivative using symmetric difference quotient (finite diff)
+            float SlopeOfAngleFunction(float input)
+            {
+                var angleBehind = AngleFunction(input - epsilon);
+                var angleAfter = AngleFunction(input + epsilon);
+                return (angleAfter - angleBehind) / (2f * epsilon);
+            }
+            // initial guess based on closest line (approximating spline) to point 
+            float InitialGuess()
+            {
+                UpdateCachedSpline();
+                var pb = m_CachedKnots[1]; // point at the bottom of spline
+                var pm = m_CachedKnots[2]; // point in the middle of spline
+                var pt = m_CachedKnots[3]; // point at the top of spline
+                var dbm = SqrDistanceBetweenLineSegmentAndPoint(pb, pm, cameraPos, out var tbm);
+                var dmt = SqrDistanceBetweenLineSegmentAndPoint(pm, pt, cameraPos, out var tmt);
+
+                var mid = (min + max) / 2f;
```

Comment:
style issue: a little weird to be using min and max here.  Why are you using them?